### ISSUE TITLE
"cargo owner --add" doesn't seem to work with teams

### ISSUE THREAD

**Issue by jntrnr** [rust-lang/cargo#11505]
### Problem

We've tried a few different ways of adding a team to grant the publishing rights to Nushell's repo, but haven't been able for this to work.

Here's what we're seeing:

```
〉 cargo owner --add github:nushell:publishing
error: failed to invite owners to crate `nu` on registry at https://crates.io

Caused by:
  the remote server responded with an error: could not find the github team nushell/publishing
```

### Steps

1. see above

### Possible Solution(s)

It'd be nice to perhaps give a better error to help add a team if there's an issue with adding the specific team. Otherwise, I think this should work based on the name of the org + team.

### Version

```text
cargo 1.66.0 (d65d197ad 2022-11-15)
release: 1.66.0
commit-hash: d65d197ad5c6c09234369f219f943e291d4f04b9
commit-date: 2022-11-15
host: aarch64-apple-darwin
libgit2: 1.5.0 (sys:0.15.0 vendored)
libcurl: 7.84.0 (sys:0.4.55+curl-7.83.1 system ssl:(SecureTransport) LibreSSL/3.3.6)
os: Mac OS 13.0.1 [64-bit]
```

**Comment by weihanglo** [comment]
Sorry about that. Looks like you hit this [issue](https://doc.rust-lang.org/nightly/cargo/reference/publishing.html#troubleshooting-github-team-access-errors). Have you tried following that steps?

**Comment by jntrnr** [comment]
Yes, I think that ultimately was it. I had to revoke permissions, then have crates.io ask for them again after revoking them (log out then log back in, and then it asks), then make sure that nushell was in the grant for the permissions it asked for.

Something that would help a lot here is to show some screenshots of the github and crates.io side. I read through the text a bunch but it didn't click what exactly was meant since I kept seeing crates.io listed in the oauth apps that had permission. It was only after the 5th read it finally hit me that it was too old to have the proper permissions granted.

**Comment by VincentWo** [comment]
I could take over this issue if wanted :)

**Comment by weihanglo** [comment]
@VincentWo go ahead! I guess adding one or two screenshots as what JT said [under this paragraph](https://github.com/rust-lang/cargo/blob/master/src/doc/src/reference/publishing.md?plain=1#L245) is sufficient. You may need an org account to reproduce the error.

**Comment by VincentWo** [comment]
Will do - thank you for the pointer.

But I wonder whether it would also be a goal to improve the error message

**Comment by weihanglo** [comment]
Oh yup. Nice suggestion! We could also provide a link to <https://doc.rust-lang.org/cargo/reference/publishing.html#github-permissions> in the error message.

**Comment by torhovland** [comment]
As discussed in #12511, I have opened a PR against crates.io here: https://github.com/rust-lang/crates.io/pull/8593

**Comment by torhovland** [comment]
I am actually not sure if [the documentation](https://github.com/rust-lang/cargo/blob/master/src/doc/src/reference/publishing.md?plain=1#L245) is wrong or incomplete. Somebody with a GitHub org account should try to first login with a key without [the read:org scope](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/), and then figure out what needs to be cleaned up in order to get things working.

**Comment by epage** [comment]
It looks like rust-lang/crates.io#8593 resolved this, so I'm going to close this. If there is something that is missing with that, let us know!
